codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -422,6 +415,9 @@ public void error(Throwable cause) {
             lifecycleManager.onException(ctx, cause);
             data = null;
             size = 0;
+            // Remove the listener in order to
+            // not call error() twice.
+            promise.removeListener(this);
```

Comment:
If we accept that we can call `error` on ourselves (via the promise failing), maybe it's ok.